File: celladapter/src/main/java/io/erva/celladapter/x/select/mode/SingleSelectionManager.java
package io.erva.celladapter.x.select.mode;

public class SingleSelectionManager extends SelectionManager {

    @Override
    public void toggleSelection(Integer position) {
        clearSelections(true);
        super.toggleSelection(position);
        adapter.notifyItemChanged(position);
    }

    @Override
    public void setSelection(Integer position, boolean isSelected) {
        clearSelections(true);
        super.setSelection(position, isSelected);
        adapter.notifyItemChanged(position);
    }

    @Override
    public boolean isSelected(int position) {
        return getSelectedPositions().contains(position);
    }

    public Integer getSelectedPosition() {
        return getSelectedPositions().isEmpty()? -1 : getSelectedPositions().iterator().next();
    }
}

File: celladapter-kotlin/src/main/kotlin/io/erva/celladapter/select/mode/SingleSelectionManager.kt
package io.erva.celladapter.select.mode

class SingleSelectionManager : SelectionManager() {

    override fun toggleSelection(position: Int) {
        clearSelections(true)
        super.toggleSelection(position)
        adapter.notifyItemChanged(position)
    }

    override fun setSelection(position: Int, isSelected: Boolean) {
        clearSelections(true)
        super.setSelection(position, isSelected)
        adapter.notifyItemChanged(position)
    }

    override fun isSelected(position: Int): Boolean {
        return getSelectedPositions().contains(position)
    }

    fun getSelectedPosition(): Int? {
        return if (getSelectedPositions().isEmpty()) -1 else getSelectedPositions().iterator().next()
    }
}